src/main/java/uk/ac/ic/wlgitbridge/bridge/gc/GcJobImpl.java

```diff
@@ -118,16 +116,25 @@ private void doGC() {
         ) {
             String proj = it.next();
             Log.info("[{}] Running GC job on project", proj);
-            try (LockGuard __ = locks.lockGuard(proj)) {
+            ProjectContext context = ContextStore.Instance().getContextForProject(proj);
+            ProjectLock lock = context.getLock();
+            try {
+                lock.lock();
```

Comment:
nitpick: This `lock()` is within the `try` block, where elsewhere we have called it before the `try`.

In theory, the Postgres lock could throw an exception, and in fact as there are two SQL operations in there we could end up succeeding on one and failing on the other - meaning that we leave a connection that hasn't had its transaction properly closed or failed.

I think we should probably standardise on putting `lock` within the `try`. WDYT?

Reply:
Agreed :+1: